src/modm/driver/radio/dw3110/dw3110_phy.hpp

```diff
@@ -0,0 +1,472 @@
+	getFilterConfig();
+
+	/// Set frame filtering config
+	modm::ResumableResult<void>
+	setFilterConfig(Dw3110::FilterConfig_t fc);
+
+	/// Enable auto acknowledgment \n
+	/// Will automatically respond to valid 802.5.14 Frames with the ACK request
+	/// bit set \n
+	/// Only active when Frame filtering is enabled
+	modm::ResumableResult<void>
+	setAutoAckEnabled(bool value);
+
+	/// Read the reported state of the chip
+	modm::ResumableResult<Dw3110::SystemState>
+	getChipState();
+
+	/// Set both the RX and TX antenna delay \n
+	/// Unit is approx 15.65ps
+	modm::ResumableResult<void>
+	setAntennaDelay(uint16_t delay);
+
+	/// Get the currently programmed RX antenna delay \n
+	/// Unit is approx 15.65ps
+	modm::ResumableResult<uint16_t>
+	getRXAntennaDelay();
+
+	/// Get the currently programmed TX antenna delay \n
+	/// Unit is approx 15.65ps
+	modm::ResumableResult<uint16_t>
+	getTXAntennaDelay();
+
+	/// Get the Clock offset multiplier for the received packet \n
+	/// Can be used to recover the clock drift compared to a packets sender \n
+	/// Unit is in parts per million (ppm)
+	modm::ResumableResult<float>
+	getReceiverClockOffset();
+
+	/// Runs the RX calibration. \n
+	/// Needs to be done after startup, wake and after 20C temperature change \n
+	/// Needs to be run when chip is in IDLE_PLL state \n
+	/// Automatically run by @ref initialize()
+	modm::ResumableResult<bool>
+	calibrate();
+
+	/// Initialize the DW3000, make ready to receive/transmit \n
+	/// Call ideally after the IRQ Pin has gone high, signalling SPIRDY
+	modm::ResumableResult<bool>
+	initialize(Dw3110::Channel channel = Dw3110::Channel::Channel5,
+			   Dw3110::PreambleCode pcode = Dw3110::PreambleCode::Code_64Mhz_9,
+			   Dw3110::PreambleLength plen = Dw3110::PreambleLength::Preamble_64,
+			   Dw3110::StartFrameDelimiter sfd = Dw3110::StartFrameDelimiter::IEEE802_15_4z_8);
+
+	/// Set the chip into receive mode
+	modm::ResumableResult<bool>
+	startReceive();
+
+	/// Check if a packet has been successfully received
+	modm::ResumableResult<bool>
+	packetReady();
+
+	/// Check if the chip is currently in RX mode
+	modm::ResumableResult<bool>
+	isReceiving();
+
+	/// Copy received packet into the provided payload buffer, clear packet
+	/// received flags
+	/// @param payload Region of memory the payload will be written to
+	/// @param payload_len Contains the length of the received payload
+	modm::ResumableResult<bool>
+	fetchPacket(std::span<uint8_t> payload, size_t &payload_len);
+
+	/// Transmit a given package using the current configuration using a given transmission mode
+	/// @param payload Span to the desired payload
+	/// @param ranging Decides whether or not to set the ranging bit in the header
+	/// @param fast Decides if the data portion is sent at 850kbps or 6.8Mbps
+	template<TXMode tmode = TXMode::Default>
+	modm::ResumableResult<Error>
+	transmit(const std::span<const uint8_t> payload, bool ranging = true, bool fast = true);
```

Comment:
Just fyi, the first `const` here does nothing. `const`-qualifying a pass-by-value argument in a function declaration has no effect. In the definition, though, it makes the span itself `const`, e.g. you can't reassign what the span points to.
It is sufficient to have `std::span<const uint8_t>` to make the data `const`. The way it's used here looks unintentional because it's not consistent with all the other value arguments which aren't spans being non-`const`.